cmd/guacone/cmd/osv.go

```diff
@@ -92,45 +98,67 @@ var osvCmd = &cobra.Command{
 			return nil
 		}
 
-		// Collect
-		errHandler := func(err error) bool {
-			if err == nil {
-				logger.Info("certifier ended gracefully")
-				return true
+		operation := func() error {
+			ctx, cf := context.WithCancel(ctx)
+			defer cf() // Ensure to cancel the context when we're done.
+
+			var wg sync.WaitGroup
+			done := make(chan bool, 1)
+			gotErr := false // Track if we encounter an error.
+
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				errHandler := func(err error) bool {
+					if err == nil {
+						logger.Info("certifier ended gracefully")
+						return true
+					}
+					logger.Errorf("certifier ended with error: %v", err)
+					gotErr = true
+					return true
+				}
+
+				if err := certify.Certify(ctx, packageQuery, emit, errHandler, opts.poll, opts.interval); err != nil {
+					logger.Errorf("Unhandled error in the certifier: %s", err)
+					gotErr = true
+				}
+				done <- true
+			}()
+
+			sigs := make(chan os.Signal, 1)
+			signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
i think we don't want to call the syscall in a loop for every operation, we will want to only do this once.